core/templates/pages/topic-editor-page/modal-templates/questions-opportunities-select-difficulty-modal.component.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for questions opportunities select difficulty modal.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { AppConstants } from 'app.constants';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { SkillBackendApiService } from 'domain/skill/skill-backend-api.service';
+import { SkillDifficulty } from 'domain/skill/skill-difficulty.model';
+import { Skill } from 'domain/skill/SkillObjectFactory';
+import { ExtractImageFilenamesFromModelService } from 'pages/exploration-player-page/services/extract-image-filenames-from-model.service';
+import { AlertsService } from 'services/alerts.service';
+import { AssetsBackendApiService } from 'services/assets-backend-api.service';
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+
+@Component({
+  selector: 'oppia-questions-opportunities-select-difficulty-modal',
+  templateUrl:
+    './questions-opportunities-select-difficulty-modal.component.html'
+})
+export class QuestionsOpportunitiesSelectDifficultyModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+  @Input() skillId: string;
+  instructionMessage: string;
+  skillIdToRubricsObject: object;
```

Comment:
Can we use a more accurate type here?

Reply:
It is declared as an object in topic-editor-state.service (See line no 58, 323) and also used as type object in all other places as well.

Reply:
Hm, could we do something like this (see type)?

https://github.com/oppia/oppia/blob/4598af0b5804aa15f96c24ce49c3c909e322de01/core/templates/components/question-difficulty-selector/question-difficulty-selector.component.ts#L36-L37

Reply:
Done. Thanks!

Reply:
Could you also please create an issue to fix the type in other places? Thanks

Reply:
Okay sure, will do that.